app/src/main/java/com/lightricks/feedexercise/data/FeedRepository.kt

```diff
@@ -1,9 +1,48 @@
 package com.lightricks.feedexercise.data
 
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Transformations
+import com.lightricks.feedexercise.database.AppDatabase
+import com.lightricks.feedexercise.database.FeedItemEntity
+import com.lightricks.feedexercise.network.FeedApiService
+import com.lightricks.feedexercise.network.FeedItemDTO
+import io.reactivex.Completable
+
 /**
  * This is our data layer abstraction. Users of this class don't need to know
  * where the data actually comes from (network, database or somewhere else).
  */
-class FeedRepository {
-    //todo: implement
+class FeedRepository (
+    private val feedApiService: FeedApiService,
+    private val feedDatabase : AppDatabase
+){
+    private val feedItems: LiveData<List<FeedItem>> =
+        Transformations.map(feedDatabase.feedItemDao().getAll()) {
+            it.toFeedItems()
+        }
+
+    public fun refresh() : Completable{
+        return feedApiService.singleFeedItem().flatMapCompletable { response ->  insertToDb(response.templatesMetadata)}
+    }
+
+
+    private fun insertToDb(items: List<FeedItemDTO>?): Completable{
+        val listEntity = ArrayList<FeedItemEntity>()
```

Comment:
You can make the code nicer if you define `toFeedItemEntity()` similar to `toFeedItems()`